lib/TBDGen/TBDGen.cpp

```diff
@@ -232,14 +233,46 @@ void TBDGenVisitor::addDifferentiabilityWitness(
 
 void TBDGenVisitor::addDifferentiableAttr(AbstractFunctionDecl *original,
                                           const DifferentiableAttr *attr) {
-  addAutoDiffLinearMapFunction(original, attr,
+  addAutoDiffLinearMapFunction(original, attr->getParameterIndices(),
                                AutoDiffLinearMapKind::Differential);
-  addAutoDiffLinearMapFunction(original, attr, AutoDiffLinearMapKind::Pullback);
-  addAutoDiffDerivativeFunction(original, attr,
+  addAutoDiffLinearMapFunction(original, attr->getParameterIndices(),
+                               AutoDiffLinearMapKind::Pullback);
+  addAutoDiffDerivativeFunction(original, attr->getParameterIndices(),
                                 AutoDiffDerivativeFunctionKind::JVP);
-  addAutoDiffDerivativeFunction(original, attr,
+  addAutoDiffDerivativeFunction(original, attr->getParameterIndices(),
                                 AutoDiffDerivativeFunctionKind::VJP);
-  addDifferentiabilityWitness(original, attr);
+  addDifferentiabilityWitness(original, attr->getParameterIndices(),
+                              attr->getDerivativeGenericSignature());
+}
+
+void TBDGenVisitor::addDerivativeAttr(AbstractFunctionDecl *derivative,
```

Comment:
This this be simplified by iterating over the list of derivative configurations from the previous PR, instead of iterating over the attributes?

I'm not totally sure if we're allowed to use that information in TBDGen. I think it's allowed. For example, TBDGen uses `DC->getLocalConformances()` which uses `NominalTypeDecl::ConformanceTable`, which is a similar piece of information.

Reply:
Done in https://github.com/apple/swift/pull/28621/commits/0c531bb8f21a556e367a5206f932d8980570291f. TBDGen.cpp is simpler now!